src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,769 @@
-                preview();
-
-
-                $("#selecttime").find("option").each(function () {
-                    if ($(this).attr("data-value") === response.timezone) {
-                        $("#selecttime").val($(this).val()).trigger("change");
-                    }
-                });
-
-                var table = $("#timeTable");
-                var lines = response.dateTime.length;
-                var rows = document.getElementById("timeTable").rows.length - 1;
-                while (lines > rows) {
-                    addRow(table, "2", "20");
-                    rows++;
-                }
-                while (lines < rows) {
-                    delLastRow("timeTable");
-                    rows--;
-                }
-
-                var i = 0;
-                table.find("tbody tr").each(function () {
-                    var $td = $(this).find("td");
-                    $td.eq(0).find("input").val(response.dateTime[i].column);
-                    $td.eq(1).find("input").val(response.dateTime[i++].string);
-                });
-
-
-                lines = response.streamData.length;
-                if (lines > 0) {
-                    thingConfig(response.streamData, response.frostURL);
-                }
-            });
-           
-        },
-        error: function (e) {
-            addToLog(e.responseText);
-        }
-    });
+	$
+			.ajax({
+				type : "GET",
+				url : "config/single",
+				data : {
+					configId : id
+				},
+				success : function(response) {
+
+					mappingData = response.mapOfMagicNumbers;
+					$("#frostserverurl").val(response.frostURL);
+
+					urlconfirmed(function() {
+						$("#delimiter").val(response.delimiter);
+						$("#headerlines").val(response.numberOfHeaderlines);
+						currentDelimiter = response.delimiter;
+						currentHeaderLines = response.numberOfHeaderlines;
+						preview();
+
+						$("#selecttime")
+								.find("option")
+								.each(
+										function() {
+											if ($(this).attr("data-value") === response.timezone) {
+												$("#selecttime").val(
+														$(this).val()).trigger(
+														"change");
+											}
+										});
+
+						var table = $("#timeTable");
+						var lines = response.dateTime.length;
+						var rows = document.getElementById("timeTable").rows.length - 1;
+						while (lines > rows) {
+							addRow(table, "2", "20");
+							rows++;
+						}
+						while (lines < rows) {
+							delLastRow("timeTable");
+							rows--;
+						}
+
+						var i = 0;
+						table.find("tbody tr").each(
+								function() {
+									var $td = $(this).find("td");
+									$td.eq(0).find("input").val(
+											response.dateTime[i].column);
+									$td.eq(1).find("input").val(
+											response.dateTime[i++].string);
+								});
+
+						lines = response.streamData.length;
+						if (lines > 0) {
+							thingConfig(response.streamData, response.frostURL);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['thingConfig' was used before it was defined.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)